mythril/laser/ethereum/time_handler.py

```diff
@@ -0,0 +1,17 @@
+import time
+
+
+class TimeHandler:
+    def __init__(self):
+        self._start_time = None
+        self._execution_time = None
+
+    def start_execution(self, execution_time):
+        self._start_time = int(time.time() * 1000)
+        self._execution_time = execution_time * 1000
+
+    def time_remaining(self):
+        return self._execution_time - (int(time.time() * 1000) - self._start_time)
+
+
+time_handler = TimeHandler()
```

Comment:
What do you think about using the singleton meta class here? 

```

class Singleton(type):
    """A metaclass type implementing the singleton pattern."""

    _instances = {}

    @synchronized(lock)
    def __call__(cls, *args, **kwargs):
        """Delegate the call to an existing resource or a a new one.

        This is not thread- or process-safe by default. It must be protected with
        a lock.

        :param args:
        :param kwargs:
        :return:
        """
        if cls not in cls._instances:
            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
        return cls._instances[cls]
```

Reply:
Its currently in the signatures file, but it can/should be moved to some util's class if we use it here as well

Reply:
We don't need a lock for this module right?I have added a singleton class without any locks in the support_utils module. And also the signature's singleton uses a lock generated in that file, I don't have much idea about sending a lock to the singleton class. what do you think about looking at it in a different PR.

Reply:
Oh, I hadn't payed attention to the lock. `what do you think about looking at it in a different PR.` Yeah that's fine